src/application-binary-interface/libcaf.h

```diff
@@ -88,24 +99,12 @@ typedef enum caf_deregister_t
   CAF_DEREGTYPE_COARRAY_DEALLOCATE_ONLY
 } caf_deregister_t;
 
+/** The opaque type to represent a coarray token. */
 typedef void *caf_token_t;
-/** Add a dummy type representing teams in coarrays. */
 
+/** The opaque type for teams. */
 typedef void *caf_team_t;
 
-typedef struct caf_teams_list
-{
-  caf_team_t team;
-  int team_id;
-  struct caf_teams_list *prev;
-} caf_teams_list;
-
-typedef struct caf_used_teams_list
```

Comment:
What was the rationale for moving these out of the header file into the body of mpi_caf.c?

Reply:
The Fortran standard defines the team_type to be an opaque type, i.e., its implementation must not be exposed. Having it in the header file may allow some nosey tinkerer to fiddle with it. Hiding it in the implementation (being an implementation detail) just makes it harder for him to exploit it. 
In fact, it is just taste. I did the same with the team's type like with the token's type. Both are only defined in the implementation and the header has an opaque pointer only.